pkg/sql/randgen/expr.go

```diff
@@ -243,3 +244,24 @@ func randExpr(
 
 	return expr, typ, nullability
 }
+
+// typeToStringCastHasIncorrectVolatility returns true for a given type if the
+// cast from it to STRING types has been given an incorrect volatility. For
+// example, REGCLASS->STRING casts are immutable when they should be stable (see
+// #74286 and #74553 for more details).
+//
+// Creating computed column expressions with such a cast can cause logical
+// correctness bugs and internal errors. The volatilities cannot be fixed
+// without causing backward incompatibility, so this function is used to prevent
+// sqlsmith and TLP from repetitively finding these known volatility bugs.
+func typeToStringCastHasIncorrectVolatility(t *types.T) bool {
```

Comment:
should we also throw in TimestampFamily and DateFamily? (maybe IntervalFamily too, not sure)

Reply:
Good point. In that case we should merge this to master first then backport to the other releases. I'll close this and open a new PR.

Reply:
INTERVAL::STRING is immutable in Postgres, so I don't think we need that:

```
marcus=# CREATE TABLE t (i INTERVAL, t TEXT GENERATED ALWAYS AS (i::TEXT) STORED);
CREATE TABLE
```

Though I'm confused why this cast depends on `IntervalStyle`. It shouldn't if it's immutable.

https://github.com/cockroachdb/cockroach/blob/979d3ff35cfd6b163cb26b4c911c9bae78597d41/pkg/sql/sem/tree/cast.go#L573-L574

Reply:
> INTERVAL::STRING is immutable in Postgres

:blob_eyes_grin: